sda-commons-server-testing/build.gradle

```diff
@@ -11,7 +11,11 @@ dependencies {
   compile 'jakarta.servlet:jakarta.servlet-api'
 
   compile 'junit:junit'
+  compile 'org.junit.jupiter:junit-jupiter-api'
   compile 'org.hamcrest:hamcrest'
   compile 'org.mockito:mockito-core'
   compile 'org.assertj:assertj-core'
+
+  runtimeOnly 'org.junit.jupiter:junit-jupiter-engine'
+  runtimeOnly 'org.junit.vintage:junit-vintage-engine'
```

Comment:
Do we need vintage in runtime scope or should we leave it to the consumers if they want to support vintage in their tests? I guess, finally projects should decide if they use Junit4 or Junit5 but should not mix. Or is it better to provide everything that may be needed?

Reply:
we need it here. without vintage, the junit4 tests are not executed.
vice versa: without jupiter junit5 tests are not executed

Reply:
So for vintage `testCompile` should be fine, but we have to decide if we wan't to provide it for consumers.
@Fox32, @christopher-cudennec what do you think?

Reply:
I think it's common practice to not include the engine itself as compile scope dependency but to define it as `runtimeOnly`. All the projects that want to use need to add the engine themselves AFAIK.

Reply:
Ah, Ok. runtime scope is quite tricky.
`runtimeOnly` or the Maven result `runtime` [defines dependencies for execution](http://andresalmiray.com/maven-scopes-vs-gradle-configurations/#:~:text=Maven%20defines%206%20scopes%3A%20compile,system%2C%20test%2C%20and%20import.&text=runtime%20This%20scope%20indicates%20that,but%20not%20the%20compile%20classpath.). That lead to my idea that it affects the consumers. But it also means that [the dependency is not inherited](https://stackoverflow.com/a/12273576).

Should we add a general part in the README about what is needed to use Junit5?

Reply:
We've just double checked one consumer: It's not necessary to add the engine dependency because you already get it from "sda-commons-server-testing". You only need to activate Junit platform for the test plugin in the build.gradle.
We'll add some documentation to the README 👍

Reply:
> So for vintage testCompile should be fine, but we have to decide if we wan't to provide it for consumers.
@Fox32, @christopher-cudennec what do you think?

I think the current setup is fine. Although the engines will be passed down to consumers with "test runtime" scope, tests will still be executed the known way with JUnit 4. JUnit 5 execution will only be activated by explicitly configuring the "test" plugin of your project.